src/main/java/hibernate/dao/impl/TicketDaoImpl.java

```diff
@@ -0,0 +1,98 @@
+package hibernate.dao.impl;
+
+import hibernate.dao.TicketDao;
+import hibernate.model.Ticket;
+import lombok.AllArgsConstructor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.MutationQuery;
+import org.hibernate.query.Query;
+import ticket.model.TicketType;
+
+@AllArgsConstructor
+public class TicketDaoImpl implements TicketDao {
+
+    private static final String FIND_BY_USER_ID = "FROM Ticket t WHERE t.user.id = :userId";
+    private static final String UPDATE_TICKET_TYPE = "UPDATE Ticket t SET t.ticketType = :ticketType WHERE t.id = :id";
+
+    private final SessionFactory sessionFactory;
+
+    @Override
+    public void save(Ticket ticket) {
+        try (Session session = sessionFactory.openSession()) {
+            try {
+                session.beginTransaction();
+                session.persist(ticket);
+                session.getTransaction().commit();
+            } catch (Exception e) {
+                if (session.getTransaction() != null && session.getTransaction().isActive()) {
+                    session.getTransaction().rollback();
+                }
+                throw new RuntimeException("Error while saving ticket", e);
+            }
+        }
+    }
+
+
+    @Override
+    public Ticket findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            try {
+                session.beginTransaction();
+                Ticket ticket = session.get(Ticket.class, id);
+                session.getTransaction().commit();
+                return ticket;
```

Comment:
you don't need to commit here, because e just read daya))
so can be simplify like
return session.get(Ticket.class, id);